### PR TITLE
[FIRRTL] Innocuous change to mitigate #6513

### PR DESCRIPTION
**Pull request by teqdruid**
This change seems to make the test problem go away. Care of @fzi-hielscher in #6570 .

### CODE REVIEW

**Comment by teqdruid** [comment]
"Fixed" the problem locally. Testing build: https://github.com/llvm/circt/actions/runs/7483263155.

**Comment by fzi-hielscher** [comment]
Thank you. I'll admit I have no clue what's going on there and this is just the result of me poking the problem.
For reference: I tested it using Visual Studio Version 17.8.4, `cl` version 19.38.33134 x64 on the `RelWithDebInfo` configuration.

File: lib/Dialect/FIRRTL/Transforms/Dedup.cpp [hunk 1]
```diff
@@ -1399,7 +1399,8 @@ void fixupAllModules(InstanceGraph &instanceGraph) {
         continue;
       ImplicitLocOpBuilder builder(inst.getLoc(), inst->getContext());
       builder.setInsertionPointAfter(inst);
-      for (unsigned i = 0, e = getNumPorts(module); i < e; ++i) {
+      auto e = getNumPorts(module);
+      for (unsigned i = 0; i < e; ++i) {
```
Comment (seldridge) [inline lib/Dialect/FIRRTL/Transforms/Dedup.cpp:1403]: image

I'd be surprised if this is actually fixing it and just permuting things enough that it works. 🤔 Anyway, go for it and see where we get. Thanks for filing this.

Comment (teqdruid) [reply lib/Dialect/FIRRTL/Transforms/Dedup.cpp:1403]: I said pretty much the same thing over in #6513.

**Pull request merged**